### PR TITLE
docs: Revise last section of spec contents (4/4)

### PR DESCRIPTION
**Pull request by juliaogris**
Revise spec contents on operators, whitespace and functions. The context of
these changes is as follows:

✅ Intro
✅ Evy syntax specification
✅ Comments
✅ Types
✅ Variables and Declarations
✅ Zero Values
✅ Assignments
✅ Copy and reference
✅ Scope
✅ Strings
✅ Arrays
✅ Maps
✅ Index and Slice
✅ Operators and Expressions
✅ Precedence
✅ Whitespace
✅ Functions
✅ Variadic functions
👉 Break and Return
👉 Typeof
👉 Type Assertion
👉 Run-time Panics and Recoverable Errors
👉 Execution Model and Event Handler
👉 Runtimes

---

Review link: https://github.com/foxygoat/evy/blob/spec-revision-4/docs/spec.md#break-and-return

After this PR I want to create one more cleanup PR replacing italics with
bold for technical terms. This leaves italics for emphasising ordinary words
and seems to better highlight the technical terms.

I also want to consistently title-case or not title case titles.

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -1135,85 +1136,252 @@ in Evy. The call arguments must be listed individually.
...
 
-`typeof` returns the concrete type of a value held by a variable as a
-string. It returns a string the same as the type in an evy program,
-e.g. `num`, `bool`, `string`, `[]num`, `{}[]any`, etc. For an empty
-composite literal , `typeof` returns `[]` or `{}` as it can be matched
-to any subtype. e.g. `[]` can be passed to a function that takes an
-argument of `[]num`, or `[]string`, etc.
+The `typeof` function returns the concrete type of a value held by a
+variable as a string. It returns a string that is the same as the type
+in an Evy program, such as `"num"`, `"bool"`, `"string"`, `"[]num"`, `"
```
Comment (camh-) [inline docs/spec.md:1209]: putting a line break within the backticks causes a space to go into the text (`" {}[]any"`). I would move the line break before the opening backtick if you can - not sure if prettier is going to just move that again though.

File: docs/spec.md [hunk 2]
```diff
@@ -1135,85 +1136,253 @@ in Evy. The call arguments must be listed individually.
...
+
+typeof x: any
+typeof str: string
+```
 
 Only values of type `any` can be type asserted. That means an array of
-type any, `[]any`, _cannot_ be type assert to be an array of type `num`
-or other concrete type:
+type any, `[]any`, _cannot_ be type asserted to be an array of type
+`num` or any other concrete type. However, the elements of an array of
```
Comment (camh-) [inline docs/spec.md:1308]: `num` or `[]num` here? earlier in the sentence you say "an array of type any, `[]any`", then you say "an array of type `num`".

**Review by camh- — APPROVED** [review]
:broccoli: with some small comments

File: docs/spec.md [hunk 3]
```diff
@@ -1135,85 +1136,253 @@ in Evy. The call arguments must be listed individually.
...
+no break 0
+no break 1
+break 2
+x 1 y 2
+
+```
+
+As you can see, the `break` statement causes the loop to exit when the
+value of `y` is equal to 2. The next statement after the loop is then
+executed.
```
Comment (camh-) [inline docs/spec.md:1184]: Since you used nested loops in the example, perhaps it deserves a mention in the text?

File: docs/spec.md (hunk 1 above)
Comment (juliaogris) [reply docs/spec.md:1209]: sorry, fixed.

File: docs/spec.md (hunk 2 above)
Comment (juliaogris) [reply docs/spec.md:1308]: yes, changed to `[]num`.

File: docs/spec.md (hunk 3 above)
Comment (juliaogris) [reply docs/spec.md:1184]: yeah, good point. I've mentioned it in the text now.

**Pull request merged**